### PR TITLE
task/DES-2759 : Apps side nav styling updates

### PR DESCRIPTION
**Pull request by edmondsgarrett**
## PR Status: ##

* [X] Ready.
* [ ] Work in Progress.
* [ ] Hold.

## Related Jira tickets: ##

* [DES-2759](https://tacc-main.atlassian.net/browse/DES-2759)

## Summary of Changes: ##
- Applied active and hover background color for nav elements via antd's Menu theme tokens
- Added `shortLabel` as first option for rendering an app's label in the side nav

## Testing Steps: ##
1. Go to https://designsafe.dev/rw/workspace/padcirc?appVersion=55.01 (or any app page, as convenient)
2. Confirm the hover + selected color for nav items matches the screenshot below
3. Confirm there is now a dark border on the right side of the side nav as shown in the screenshot below
4. Confirm that, in side nav, for the ADCIRC (Frontera) app, only "ADCIRC" is shown

## UI Photos:
image

## Notes: ##
Please review code additions relating to rendering front end - unsure what I am missing to capture the shortLabel for the adcirc-frontera app. _Edit: fix added in commit [2031df9](https://github.com/DesignSafe-CI/portal/pull/1280/commits/2031df9990656ea419dc34db2777cb29956fea16)_

### CODE REVIEW

File: designsafe/apps/workspace/models/app_entries.py [hunk 1]
```diff
@@ -176,6 +176,12 @@ class AppVariant(models.Model):
         blank=True,
     )
 
+    shortLabel = models.CharField(
+        help_text="The display name of this app in the Apps side navigation. If not defined, uses notes.shortLabel from app definition.",
```
Comment (rstijerina) [inline designsafe/apps/workspace/models/app_entries.py:180]: Can you make a second PR to main with this additional field and migration?

File: client/modules/workspace/src/AppsSideNav/AppsSideNav.tsx [hunk 2]
```diff
@@ -9,6 +9,15 @@ export const AppsSideNav: React.FC<{ categories: TAppCategory[] }> = ({
...
 
+  const AppMenuItemTheme: ThemeConfig = {
+    components: {
+      Menu: {
+        itemHoverBg: '#cbdded',
+        itemSelectedBg: '#cbdded',
+      },
+    },
+  };
+
```
[the remark this answers is not in the document]
Comment (edmondsgarrett) [reply client/modules/workspace/src/AppsSideNav/AppsSideNav.tsx]: Great idea, modified that themeConfig with the updated color for hover + select

File: designsafe/apps/workspace/models/app_entries.py (hunk 1 above)
Comment (edmondsgarrett) [reply designsafe/apps/workspace/models/app_entries.py:180]: Elected to keep this model kept to this PR, will rely on `notes.shortLabel` for prod apps in the interim

File: designsafe/apps/workspace/api/views.py [hunk 3]
```diff
@@ -307,7 +307,9 @@ def _get_valid_apps(self, tapis_apps, portal_apps):
                 valid_tapis_apps.append(
                     {
                         **portal_app,
-                        "label": portal_app["label"] or matching_app.notes.label,
+                        "label": portal_app["label"] or matching_app.notes.get("label"),
+                        "shortLabel": portal_app["label"]
```
Comment (rstijerina) [inline designsafe/apps/workspace/api/views.py:311]:
```suggestion
                        "shortLabel": portal_app["short_label"]
```

**Pull request merged**